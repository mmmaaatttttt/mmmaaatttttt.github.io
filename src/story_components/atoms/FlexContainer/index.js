import styled, { css } from "styled-components";
import PropTypes from "prop-types";

const main = props => (props.column ? "align-items" : "justify-content");
const cross = props => (props.column ? "justify-content" : "align-items");

const FlexContainer = styled.div`
  display: flex;
  flex: ${props => props.flex};
  flex-wrap: ${props => (props.shouldWrap ? "wrap" : "nowrap")};
    ${props =>
      props.column &&
      css`
        flex-direction: column;
      `}
    ${main}: ${props => props.main};
  ${cross}: ${props => props.cross};
  margin: ${props => props.margin};
  width: ${props => props.width};
  text-align: ${props => props.textAlign};
`;

const flexAlignments = [
  "flex-start",
  "flex-end",
  "center",
  "baseline",
  "stretch",
  "space-around",
  "space-between",
  "space-evenly"
];

FlexContainer.propTypes = {
  column: PropTypes.bool.isRequired,
  cross: PropTypes.oneOf(flexAlignments).isRequired,
  flex: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  main: PropTypes.oneOf(flexAlignments).isRequired,
  margin: PropTypes.string.isRequired,
  textAlign: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  shouldWrap: PropTypes.bool.isRequired
};

FlexContainer.defaultProps = {
  column: false,
  cross: "stretch",
  flex: 1,
  main: "stretch",
  margin: "0",
  textAlign: "left",
  width: "auto",
  shouldWrap: false
};

export default FlexContainer;
