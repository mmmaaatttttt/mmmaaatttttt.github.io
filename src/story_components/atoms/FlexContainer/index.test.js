import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import FlexContainer from ".";

const wrap = (props = {}) => shallow(<FlexContainer {...props} />);

it("renders successfully", () => {
  const wrapper = wrap();
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders props when passed in", () => {
  const wrapper = wrap({
    column: true,
    wrap: true,
    main: "center",
    cross: "center"
  });
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("renders children when passed in successfully", () => {
  const wrapper = shallow(
    <FlexContainer column>
      <div>Here's a child</div>
      <div>Here's another</div>
      <div>Here's a third</div>
    </FlexContainer>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
