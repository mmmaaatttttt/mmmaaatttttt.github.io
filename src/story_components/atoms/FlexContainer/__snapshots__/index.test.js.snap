// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children when passed in successfully 1`] = `
<StyledComponent
  column={true}
  cross="stretch"
  flex={1}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex: ",
          [Function],
          ";
  flex-wrap: ",
          [Function],
          ";
    ",
          [Function],
          "
    ",
          [Function],
          ": ",
          [Function],
          ";
  ",
          [Function],
          ": ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  width: ",
          [Function],
          ";
  text-align: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "column": false,
        "cross": "stretch",
        "flex": 1,
        "main": "stretch",
        "margin": "0",
        "textAlign": "left",
        "width": "auto",
        "wrap": false,
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "column": [Function],
        "cross": [Function],
        "flex": [Function],
        "main": [Function],
        "margin": [Function],
        "textAlign": [Function],
        "width": [Function],
        "wrap": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  main="stretch"
  margin="0"
  textAlign="left"
  width="auto"
  wrap={false}
>
  <div>
    Here's a child
  </div>
  <div>
    Here's another
  </div>
  <div>
    Here's a third
  </div>
</StyledComponent>
`;

exports[`renders props when passed in 1`] = `
<StyledComponent
  column={true}
  cross="center"
  flex={1}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex: ",
          [Function],
          ";
  flex-wrap: ",
          [Function],
          ";
    ",
          [Function],
          "
    ",
          [Function],
          ": ",
          [Function],
          ";
  ",
          [Function],
          ": ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  width: ",
          [Function],
          ";
  text-align: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "column": false,
        "cross": "stretch",
        "flex": 1,
        "main": "stretch",
        "margin": "0",
        "textAlign": "left",
        "width": "auto",
        "wrap": false,
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "column": [Function],
        "cross": [Function],
        "flex": [Function],
        "main": [Function],
        "margin": [Function],
        "textAlign": [Function],
        "width": [Function],
        "wrap": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  main="center"
  margin="0"
  textAlign="left"
  width="auto"
  wrap={true}
/>
`;

exports[`renders successfully 1`] = `
<StyledComponent
  column={false}
  cross="stretch"
  flex={1}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex: ",
          [Function],
          ";
  flex-wrap: ",
          [Function],
          ";
    ",
          [Function],
          "
    ",
          [Function],
          ": ",
          [Function],
          ";
  ",
          [Function],
          ": ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  width: ",
          [Function],
          ";
  text-align: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "column": false,
        "cross": "stretch",
        "flex": 1,
        "main": "stretch",
        "margin": "0",
        "textAlign": "left",
        "width": "auto",
        "wrap": false,
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "column": [Function],
        "cross": [Function],
        "flex": [Function],
        "main": [Function],
        "margin": [Function],
        "textAlign": [Function],
        "width": [Function],
        "wrap": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  main="stretch"
  margin="0"
  textAlign="left"
  width="auto"
  wrap={false}
/>
`;
